include/itkBackscatterImageFilter.h

```diff
@@ -0,0 +1,154 @@
+ *
+ * BackscatterImageFilter receives an input vector image representing
+ * RF spectra. One image direction represents the direction of an
+ * RF waveform emitted from an ultrasound probe. Remaining image
+ * directions may represent directions in physical space, such as
+ * where the probe contains multiple elements, or directions in
+ * time as multiple samples are captured in a sweep with the probe.
+ * Each pixel in the input image is a vector representing frequency
+ * components at bins based on the sampling frequency.
+ *
+ * BackscatterImageFilter generates a scalar output image with
+ * intensities representing backscatter estimates.
+ *
+ * There are three types of estimates available, see documentation for
+ * EstimateType for detailed description.
+ *
+ * \sa MaskedImageToHistogramFilter
+ * \sa Spectra1DImageFilter
+ * \sa VariableLengthVector
+ * \sa VectorImage
+ *
+ * \ingroup ITKImageStatistics
+ * \ingroup Ultrasound
+ */
+template <typename TInputImage,
+          typename TOutputImage = Image<float, TInputImage::ImageDimension>>
+class ITK_TEMPLATE_EXPORT BackscatterImageFilter : public ImageToImageFilter<TInputImage, TOutputImage>
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(BackscatterImageFilter);
+
+  /** Standard class type aliases. */
+  using Self = BackscatterImageFilter;
+  using Superclass = ImageToImageFilter<TInputImage, TOutputImage>;
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(BackscatterImageFilter, ImageToImageFilter);
+
+  /** Image type alias support */
+  static constexpr unsigned int ImageDimension = TInputImage::ImageDimension;
+
+  using InputImageType = TInputImage;
+  using InputImagePointer = typename TInputImage::Pointer;
+
+  using OutputImageType = TOutputImage;
+  using OutputImagePointer = typename TOutputImage::Pointer;
+
+  using InputRegionType = typename TInputImage::RegionType;
+  using InputSizeType = typename TInputImage::SizeType;
+  using InputIndexType = typename TInputImage::IndexType;
+  using InputPixelType = typename TInputImage::PixelType;
+
+  using OutputRegionType = typename TOutputImage::RegionType;
+  using OutputPixelType = typename TOutputImage::PixelType;
+
+  /** RF sampling frequency */
+  itkSetMacro(SamplingFrequencyMHz, float);
+  itkGetConstMacro(SamplingFrequencyMHz, float);
+
+  /** Low end of RF frequency band. Must be a positive value. */
+  itkSetMacro(FrequencyBandStartMHz, float);
+  itkGetConstMacro(FrequencyBandStartMHz, float);
+
+  /* High end of RF frequency band. Must be a positive value.*/
+  itkSetMacro(FrequencyBandEndMHz, float);
+  itkGetConstMacro(FrequencyBandEndMHz, float);
+
+  /** Estimate Type (0=average, 1=slope, 2=intercept).
+   *
+   * 0. (Default) Average of intensities of selected frequency components.
```

Comment:
I didn't think of this.

Reply:
@dzenanz How long does the least squares solver typically take to run in https://github.com/KitwareMedical/ITKUltrasound/pull/217/files#diff-a2498f24fad0d8c520b29216a685c1094870fe58bf963d4f4d881e6138ac854bR130 ?

If it is not prohibitive, one approach could be to create a separate output metric image for each enumerated `EstimateType` value.

https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkProcessObject.h#L211

Reply:
It is fast:
```log
$ ctest -C RelWithDebInfo -R Backscatter
Test project C:/Dev/ITKUltrasound-22
    Start 10: itkBackscatterImageFilterTest0_100
1/7 Test #10: itkBackscatterImageFilterTest0_100 ...   Passed    0.09 sec
    Start 11: itkBackscatterImageFilterTest0_16
2/7 Test #11: itkBackscatterImageFilterTest0_16 ....   Passed    0.08 sec
    Start 12: itkBackscatterImageFilterTest1
3/7 Test #12: itkBackscatterImageFilterTest1 .......   Passed    0.08 sec
    Start 13: itkBackscatterImageFilterTest2
4/7 Test #13: itkBackscatterImageFilterTest2 .......   Passed    0.09 sec
    Start 46: PythonBackscatterImageFilterTest0
5/7 Test #46: PythonBackscatterImageFilterTest0 ....   Passed    7.79 sec
    Start 47: PythonBackscatterImageFilterTest1
6/7 Test #47: PythonBackscatterImageFilterTest1 ....   Passed    4.67 sec
    Start 48: PythonBackscatterImageFilterTest2
7/7 Test #48: PythonBackscatterImageFilterTest2 ....   Passed    4.70 sec
```
There are already 3 regression images, one for each `EstimateType`.

Reply:
> It is fast

Excellent!

> There are already 3 regression images

It seems like there are three _types_ of metric images, and three corresponding baseline images, but the filter stores only one output metric image in memory at a time, correct? I am proposing that all three metric images should be computed and stored concurrently in memory on a given filter run.

Reply:
Ah, now I see what you mean by separate outputs. Originally, I was outputting a 3-length vector image, which was awkward to use from Python (see https://github.com/dzenanz/ITKUltrasound/commit/d8268a77c06bbc18875d88560dadc5a9b9983d0a). I am not sure whether this would be an improvement.

Reply:
>  Originally, I was outputting a 3-length vector image

[itkCoocurrenceTextureFeaturesImageFilter](itkCoocurrenceTextureFeaturesImageFilter) does this, so there is precedent, though I agree that it's more difficult to interpret the results. In my view setting N output images is both easier and safer for the user than setting the metric enumeration before the filter runs. Using N output images would remove the need to call `Update` multiple times and would be explicitly clear about which output image describes which enumerated image.

Reply:
`itkCoocurrenceTextureFeaturesImageFilter` does exactly what I had created initially, and refactored because it is cumbersome to use from Python.

Reply:
Got it, I better understand your point now after reading back over the Python test. Since we typically write single-use filter calls in ITK Python rather than having a persistent filter object it is easier to return a single object.

In an ideal world the Pythonic interface would return a tuple of three images, one for each enumerated Output from the filter. But I imagine that is nontrivial to architect.

I agree with you that it is reasonable to streamline filtering for Python if we expect that users will mostly be using ITK Python rather than ITK C++, and also that most ITK filters have the same issue where their output is invalid if parameters are changed without `Update()` being called. Fine to leave as-is.

Reply:
@dzenanz Sorry to keep closing and re-opening this issue. I did some digging and it looks like the tuple output behavior I previously described is already implemented in ITK, though not widely used.

ITK filters with N indexed outputs will return those outputs from a snake_case call as a tuple, i.e. the following is possible:

```py
average_image, slope_image, intercept_image = itk.backscatter_image_filter(
          spectra_image,
          sampling_frequency_m_hz=60,
          frequency_band_start_m_hz=5.0,
          frequency_band_end_m_hz=20.0)
```

Implementation is here: https://github.com/InsightSoftwareConsortium/ITK/blob/master/Wrapping/Generators/Python/PyBase/pyBase.i#L397

This is an optimization and not a base requirement. If you don't have bandwidth for refactoring this could be left as an issue to revisit later.

Reply:
I am implementing this now.